python/ngen_conf/src/ngen/config/path_pair/path_pair.py

```diff
@@ -0,0 +1,277 @@
+import os
+from pathlib import Path, PosixPath, WindowsPath
+
+from .protocol import Reader, Writer, Serializer, Deserializer
+from .common import path_reader, path_writer
+from ._abc_mixins import AbstractPathPairMixin, AbstractPathPairCollectionMixin
+from ._mixins import PathPairMixin, PathPairCollectionMixin
+
+from typing import Any, Dict, Generic, Optional, Union, List
+from typing_extensions import Self
+from .typing import StrPath, T
+
+
+class PathPair(AbstractPathPairMixin[T], Path, Generic[T]):
+    """A `pathlib.Path` subclass that encapsulates some inner `T` and methods for reading, writing,
+    and de/serializing to and from a `T`. Similarly to `pathlib.Path`, creating a `PathPair` will
+    return a OS specific subclass (`PosixPathPair` or `WindowsPathPair`).
+
+    The default reader and writer implementations operate on disk. Alternative reader and writer
+    types can be provided assuming they following the `ngen.config.path_path.protocol` `Reader` and
```

Comment:
should this be `ngen.config.path_pair.protocol`?

Reply:
Yep, good catch!